tests/unit/test_data_transformer.py

```diff
@@ -0,0 +1,384 @@
+
+        Input:
+            - column_transform_info
+            - raw_column_data = np.array([0.001, 11.9999, 13.001])
+
+        Output:
+            - normalized_value (assert between -1.0, 1.0)
+              - assert approx = [0.0, -1.0, 1.0]
+            - onehot (assert that it's a one-hot encoding)
+              - assert = [[0, 1], [1, 0], [1, 0]]
+
+        Side Effects:
+            - assert predict_proba called
+            - assert np.random.choice with appropriate probabilities
+        """
+
+    def test_transform(self):
+        """Test 'transform' on a dataframe with one continuous and one discrete columns.
+
+        It should use the appropriate '_transform' type for each column and should return
+        them concanenated appropriately.
+
+        Setup:
+            - Mock _column_transform_info_list
+            - Mock _transform_discrete
+            - Mock _transform_continuous
+
+        Input:
+            - raw_data = a table with one continuous and one discrete columns.
+
+        Output:
+            - numpy array containing the transformed two columns
+
+        Side Effects:
+            - _transform_discrete and _transform_continuous should each be called once.
+        """
+        data = pd.DataFrame({
+            "x": np.array([0.1, 0.3, 0.5]),
+            "y": np.array(["yes", "yes", "no"])
+        })
+
+        transformer = DataTransformer()
+        transformer._column_transform_info_list = [
+            ColumnTransformInfo(
+                column_name="x", column_type="continuous", transform=None,
+                transform_aux=None,
+                output_info=[SpanInfo(1, 'tanh'), SpanInfo(3, 'softmax')],
+                output_dimensions=1 + 3
+            ),
+            ColumnTransformInfo(
+                column_name="y", column_type="discrete", transform=None,
+                transform_aux=None,
+                output_info=[SpanInfo(2, 'softmax')],
+                output_dimensions=2
+            )
+        ]
+
+        transformer._transform_continuous = Mock()
+        selected_normalized_value = np.array([[0.1], [0.3], [0.5]])
+        selected_component_onehot = np.array([
+            [1, 0, 0],
+            [1, 0, 0],
+            [1, 0, 0],
+        ])
+        transformer._transform_continuous.return_value = (selected_normalized_value,
+                                                          selected_component_onehot)
+
+        transformer._transform_discrete = Mock()
+        transformer._transform_discrete.return_value = [np.array([
+            [0, 1],
+            [0, 1],
+            [1, 0],
+        ])]
+
+        result = transformer.transform(data)
+        transformer._transform_continuous.assert_called_once()
+        transformer._transform_discrete.assert_called_once()
+        assert result.shape == (3, 6)
+        assert (result[:, 0] == np.array([0.1, 0.3, 0.5])).all()
+        assert (result[:, 1:4] == np.array([
```

Comment:
I suggest we use variable for the expected result on one forth, similar to this:
```python
        one_forth_expected = np.array([
            [1, 0, 0],
            [1, 0, 0],
            [1, 0, 0],
        ])
        assert (result[:, 1:4] == one_forth_expected).all()
```
It's easier to read and we don't have that multi-line.